src/encord_active/lib/premium/querier.py

```diff
@@ -0,0 +1,64 @@
+import os
+from typing import Optional, Type, TypeVar
+
+import requests
+from pydantic import BaseModel
+from requests import ConnectionError
+
+from encord_active.lib.premium.model import (
+    CLIPQuery,
+    CodeSearchResponse,
+    SearchResponse,
+    TextQuery,
+)
+from encord_active.lib.project.project_file_structure import ProjectFileStructure
+
+T = TypeVar("T", bound=BaseModel)
+
+
+class Querier:
+    def __init__(self, pfs: ProjectFileStructure):
+        self.pfs = pfs
+        self.api_url = os.getenv("PREMIUM_API_URL", "http://localhost:5051")
+        self._premium_available: Optional[bool] = None
+
+    def set_project(self, pfs: ProjectFileStructure):
+        self.pfs = pfs
+
+    def execute(
+        self, endpoint: str = "", data: Optional[dict] = None, timeout: Optional[float] = None
+    ) -> Optional[dict]:
+        params = {"project": self.pfs.project_dir.as_posix()}
+        response = requests.post(f"{self.api_url}/{endpoint}", params=params, json=data, timeout=timeout)
+        if response.status_code != 200:
+            return None
+
+        return response.json()
+
+    @property
+    def premium_available(self) -> bool:
+        if self._premium_available is None:
+            try:
+                self._premium_available = bool(self.execute(timeout=0.5))
+            except (ConnectionError, ConnectionRefusedError, Exception):
+                pass
+        return self._premium_available or False
```

Comment:
a `0.5` timeout seems reasonable in a `localhost` setup, but on the network that might be too short.

May I suggest that we rely purely on the `ENV` variable, i.e. premium is available only if env var is present, in which case we raise an error if the timeout is met. (maybe half a minute?)